### PR TITLE
Add API for editing project name

### PR DESCRIPTION
**Pull request by danielzhang130**
---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
Some api tests to come with then PR on user interface

### CODE REVIEW

File: src/Catrobat/Controller/Web/ProgramController.php [hunk 1]
```diff
@@ -334,6 +334,53 @@ public function toggleProgramVisibilityAction(string $id): Response
...
+   * @throws Exception
+   */
+  public function editProgramName(string $id, string $new_name): Response
+  {
+    $max_name_size = (int) $this->getParameter('catrobat.max_name_upload_size');
+
+    if (strlen($new_name) > $max_name_size) {
+      return Response::create(
+        $this->translator->trans('programs.tooLongName', [], 'catroweb'),
+        Response::HTTP_BAD_REQUEST
```
Comment (dmetzner) [inline src/Catrobat/Controller/Web/ProgramController.php:350]: 422 Unprocessable Entity

File: config/services.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ parameters:
   catrobat.file.storage.dir:            "%kernel.project_dir%/public/resources/programs/"
   catrobat.file.storage.path:           "resources/programs/"
   catrobat.logs.dir:                    "%kernel.project_dir%/var/log/"
+  catrobat.max_name_upload_size: 1000
```
Comment (dmetzner) [inline config/services.yaml:17]: Have you checked the app implementation? 1000 seems a lot for the project name.

@Tamdav Do you have any knowledge about this on the android side?
Or any UX recommendations? :)

File: src/Catrobat/Controller/Web/ProgramController.php [hunk 3]
```diff
@@ -334,6 +334,53 @@ public function toggleProgramVisibilityAction(string $id): Response
...
+      throw $this->createAccessDeniedException('Not your program!');
+    }
+
+    $program->setName($new_name);
+
+    $em = $this->getDoctrine()->getManager();
+    $em->persist($program);
+    $em->flush();
+
+    $extracted_file = $this->extracted_file_repository->loadProgramExtractedFile($program);
```
Comment (dmetzner) [inline src/Catrobat/Controller/Web/ProgramController.php:374]: would be nice to move this logic (Changing the name in the file.) in its own method.

**Review by dmetzner — APPROVED** [review]
The ultimate goal would be to have this logic in the public API.
http://developer.catrobat.org/Catroweb-API/

`PUT /project/{id}`

Therefore, I see no issue with merging this PR as it is.

File: config/services.yaml (hunk 2 above)
Comment (Tamdav) [reply config/services.yaml:17]: There are no restrictions for this on the android side. However, I would suggest a maximum of 50 characters, which should be quite enough for the project name.

**Pull request merged**